test/test_rpc.py

```diff
@@ -133,7 +133,12 @@ def test_reinit(self):
         store = dist.FileStore(self.file.name, self.world_size)
         dist.init_process_group(backend="gloo", rank=self.rank,
                                 world_size=self.world_size, store=store)
-        with self.assertRaisesRegex(RuntimeError, "is not unique"):
+        dist.init_model_parallel(self_name='worker{}'.format(self.rank),
```

Comment:
Shouldn't we have two separate tests here? One uses duplicate worker name and raises an error with message "is not unique" and then another initializes model_parallel again with a unique name but should raise an error with "is already initialized"?